yt/fields/field_functions.py

```diff
@@ -7,37 +7,34 @@
 
 
 def get_radius(data, field_prefix, ftype):
-    unit_system = data.ds.unit_system
-    center = data.get_field_parameter("center").in_base(unit_system.name)
-    DW = (data.ds.domain_right_edge - data.ds.domain_left_edge).in_base(
-        unit_system.name
-    )
-    # This is in cm**2 so it can be the destination for our r later.
+    center = data.get_field_parameter("center").to("code_length")
+    DW = (data.ds.domain_right_edge - data.ds.domain_left_edge).to("code_length")
+    # This is in code_length so it can be the destination for our r later.
     radius2 = data.ds.arr(
-        np.zeros(data[ftype, field_prefix + "x"].shape, dtype="float64"), "cm**2"
+        np.zeros(data[ftype, field_prefix + "x"].shape, dtype="float64"), "code_length"
     )
-    r = radius2.copy()
+
+    r = np.empty_like(radius2, subok=False)
     if any(data.ds.periodicity):
-        rdw = radius2.copy()
+        rdw = radius2.v
     for i, ax in enumerate("xyz"):
-        # This will coerce the units, so we don't need to worry that we copied
-        # it from a cm**2 array.
         np.subtract(
-            data[ftype, f"{field_prefix}{ax}"].in_base(unit_system.name),
```

Comment:
The removal of the unit conversion (i.e., `in_base(unit_system.name)`) has changed radius values for frontends where the default unit is not "code_length". I'm not sure how much performance is lost by adding it back, but I think it is necessary unless we mandate position units always come back in "code_length", which I don't think we do.